src/components/views/verification/Main.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { notification } from 'antd';
+
+import Header from './Header';
+import ConnectWallet from './ConnectWallet';
+import ConfirmProfile from './ConfirmProfile';
+import ConfirmProject from './ConfirmProject';
+import Congratulations from './Congratulations';
+import GIVIcon from '../../../assets/GIV-icon-Text.svg';
+import GIVIconFill from '../../../assets/GIV-icon-Text-fill.svg';
+import { feathersClient } from '../../../lib/feathersClient';
+import { Context as UserContext } from '../../../contextProviders/UserProvider';
+import ErrorHandler from '../../../lib/ErrorHandler';
+import config from '../../../configuration';
+import { authenticateUser, checkBalance, checkForeignNetwork } from '../../../lib/middleware';
+import ErrorPopup from '../../ErrorPopup';
+import { Context as Web3Context } from '../../../contextProviders/Web3Provider';
+import Campaign from '../../../models/Campaign';
+
+const Verification = props => {
+  const {
+    state: { web3, isForeignNetwork, balance },
+    actions: { displayForeignNetRequiredWarning },
+  } = useContext(Web3Context);
+  const {
+    state: { currentUser },
+  } = useContext(UserContext);
+
+  const { slug: projectSlug } = props.match.params;
+
+  const balanceNum = balance && balance.toNumber();
+  const isBalance = !!balanceNum;
+  const userAddress = currentUser.address;
+
+  const [step, setStep] = useState(1);
+  const [project, setProject] = useState({});
+  const [campaignSlug, setCampaignSlug] = useState({});
+  const [formIsValid, setFormIsValid] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+
+  const reportIssue = () => {
+    const body = `Issue type: User and Project Verification \nMy address: ${userAddress} \nGiveth.io Slug: ${projectSlug} \n`;
+    window.open(`${config.githubUrl}/issues/new?body=${encodeURIComponent(body)}`);
+  };
+
+  const confirmProject = (txHash, profileHash) => {
+    feathersClient
+      .service('verifiedCampaigns')
+      .create({
+        slug: projectSlug,
+        txHash,
+        url: profileHash,
+      })
+      .then(_campaign => {
+        setCampaignSlug(_campaign.slug);
+        setStep(step + 1);
+      })
+      .catch(err => {
+        if (err.message) ErrorHandler(err, err.message);
+        else ErrorHandler(err, 'Something went wrong!');
+      })
+      .finally(() => setIsSaving(false));
+  };
+
+  const createCampaignOnNetwork = () => {
```

Comment:
Like in profile section, these lines of code are copy of what is implemented in `EditCampaign` component. Please remove this redundancy by defining them in single place once.

Reply:
Done